src/Arena.js

```diff
@@ -1,554 +1,555 @@
+          <div
+            className={`opponent ${opponent.styleName}`}
+            disabled={!battleState.playerTurn}
+            onClick={myAttackRoll}
+          ></div>
         </ClickNHold>
       );
     } else {
       return (
-        <div className={`opponent ${opponent.styleName}`} disabled={!battleState.playerTurn} onClick={myAttackRoll}></div>
+        <div
+          className={`opponent ${opponent.styleName}`}
+          disabled={!battleState.playerTurn}
+          onClick={myAttackRoll}
+        ></div>
       );
     }
-  }
+  };
 
   const renderVictoryScreen = () => {
-    if (opponentHP < 1){
+    if (chosenOpponent.hp < 1) {
       return (
         <div className="overlay-container">
-            <div className="center-screen">
-              <h1>VICTORY!</h1>
-              <NavLink to="/staging"><button onClick={victory}>CONTINUE</button></NavLink>
-            </div>
-            <Route path="/staging" component={Staging}/>
+          <div className="center-screen">
+            <h1>VICTORY!</h1>
+            <NavLink to="/staging">
+              <button onClick={victory}>CONTINUE</button>
+            </NavLink>
+          </div>
+          <Route path="/staging" component={Staging} />
         </div>
       );
     }
-  }
+  };
 
   return (
-    <div className="arena-bg" >
+    <div className="arena-bg">
       {renderVictoryScreen()}
       <div className="top-center">
-        <div className="lefta stroke"><h2>Opponent: {opponent.name} </h2></div>
-        <div className="centera stroke"><h2>HP: {opponentHP} / {opponent.hp} </h2></div>
+        <div className="lefta stroke">
+          <h2>Opponent: {opponent.name} </h2>
+        </div>
+        <div className="centera stroke">
+          <h2>
+            HP: {chosenOpponent.hp} / {opponent.hp}{" "}
+          </h2>
+        </div>
       </div>
-      
+
       <div className="center-screen">
-        <div className="justify-center disable-message stroke">
-          {battleState.disableMessage}
+        {riposteAnimation && <img src={riposteImage} className="skill-animation" />}
+        <div className="justify-center disable-message stroke">{battleState.disableMessage}</div>
+        <div className="justify-center dmg-num-top stroke"></div>
+        <div className="opponent-container" id="shakeContainer">
+          {renderClickNHold()}
         </div>
-        <div className="justify-center dmg-num-top stroke">
-        </div>
-          <div className="opponent-container" id="shakeContainer">
-            {renderClickNHold()}
-          </div>
         <div className="justify-center dmg-num-bottom stroke" id="bottom-message">
           {battleState.oppHitMessage}
         </div>
         <div className="block-area"></div>
-        <img alt="weapons" src={require(`./assets/images/weapons-${myGladiator.name}.png`)} className="weapons-container" id="weaponBuff"></img>
+        <img
+          alt="weapons"
+          style={{ zIndex: 2 }}
+          src={
```

Comment:
Here's the swap of the animations that happens as fast as the react render cycle.

The reason you might have been getting bad performance is this - you were using 
`require(`some-path-${dynamic}.png`)`

Node sees a dynamic variable in require and can't cache its value to reuse, therefore does this on the fly every time it needs the value. It seems reusable to inject gladiator name there but there are downsides as wel